pkg/danm/danm_test.go

```diff
@@ -0,0 +1,38 @@
+package main
+
+import (
+  "testing"
+)
+
+var devicePool0 = "pool0"
+var devicePool1 = "pool1"
+
+var allocatedDevices = make(map[string]*[]string)
+
+func TestPopDevicePanic(t *testing.T) {
+    defer func() {
+        if r := recover(); r == nil {
+            t.Errorf("Uninitialized map should expect error.")
+        }
+    }()
+
+    popDevice(devicePool0, allocatedDevices)
+}
+
+func TestPopDevice(t *testing.T) {  
```

Comment:
however this unit testing paradigm is highly advised against, and very much not preferred in this project (I know specifically within Golang people do this, that doesn't mean much to me though :) )
packages should be tested through their external interfaces, not directly through their internals. this is why all the existing test files are actually located in a totally different package

Reply:
which might be hard to do within this specific code, and this is exactly the reason why I haven't started writing UTs for this specific file yet. needs more thorough consideration

we can live with this for the time being, again just an FYI for you regarding best practices

Reply:
for reference see the IPAM test package, and how I only test it through its Allocate, and Free calls.
this is exactly like black-box testing, just in a smaller scope